### PR TITLE
update functionality

### PR DESCRIPTION
**Pull request by sameh080081**
#add , remove & update
-here we add methods that users can add, remove and update tasks with their to-do list.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,26 +1,73 @@
...
+    });
+  });
+}
+
+function add() {
+  if (input.value !== '') {
+    array.push({
+      description: input.value,
+      completed: false,
+    });
```
Comment (Mihndim2020) [inline src/index.js]: - [x] As it is a [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md) for this project, " `By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).` ", your tasks(`todos`) are missing the index key(property). Please kindly add it, and it should be set to the value of the new array length.
- [x] Please also ensure deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @sameh080081, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Your project is running free of `Linter errors`
:heavy_check_mark: Descriptive `PR` summary

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

 #### I will leave the comments here, since your `Readme file` is not accessible for this `PR` as you did not make any change on it ####

- [OPTIONAL] Consider updating the `Getting started` section of your readme file with explicit steps to clone and run your app locally as this will be of help to non technical users.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — CHANGES_REQUESTED** [review]
Hi @sameh080081, 👋🏻👋🏻 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No Linters error 🔥🔥
- Professional READMe.md file🔥🔥 

 ## Required Changes ♻️
_Check the comments under the review._
[ REQUIRED ] please as the previous code reviewer asked make sure that you are updating the indexes after deleting 

you can check the same idea from here  so you can delete the task by the id like this

```
function deleteTaskWith(taskId) {
  arrayOfTasks = arrayOfTasks.filter((task) => task.id !== +taskId);
  addDataToLocal(arrayOfTasks);
}
```
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @sameh080081, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**